src/components/ImageView/index.tsx

```diff
@@ -1,41 +1,81 @@
+import purify from 'dompurify';
 import log from 'loglevel';
 import * as React from 'react';
 
+import { gettext } from '../../utils';
 import styles from './styles.module.scss';
 
-type PublicProps = {
+export type PublicProps = {
   _btoa?: typeof btoa;
   _log?: typeof log;
+  _sanitize?: typeof purify.sanitize;
   mimeType: string;
   content: string;
 };
 
+// From https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#Firefox.
+const supportedImageMimeTypes = [
+  'image/jpeg',
+  'image/gif',
+  'image/png',
+  'image/svg+xml',
+  'image/bmp',
+  'image/vnd.microsoft.icon',
+  // webp isn't listed in the MDN article, but is documented elsewhere.
+  // e.g., https://caniuse.com/#search=webp
+  'image/webp',
+];
+
 type ConvertImageDataParams = {
   _btoa?: typeof btoa;
   _log?: typeof log;
+  _sanitize?: typeof purify.sanitize;
   content: string;
+  mimeType: string;
 };
 
-export const convertImageData = ({
+const convertImageData = ({
   _btoa = btoa,
   _log = log,
+  _sanitize = purify.sanitize,
   content,
+  mimeType,
 }: ConvertImageDataParams): string | null => {
-  try {
-    return _btoa(content);
-  } catch (error) {
-    _log.debug('Error converting image data to ascii:', error);
+  const mimeTypeCased = mimeType.toLowerCase();
+  if (!supportedImageMimeTypes.includes(mimeTypeCased)) {
+    _log.error('Unrecognized mimeType :', mimeType);
```

Comment:
Since this message will be mixed with all other Sentry messages, it would help to make it specifically about images